In [97]:
import numpy as np
import pandas as pd, openpyxl
import statistics

# 3.2 Sample mean

1. The following data represent the scores on a statistics examination of a sample of students:  
87, 63, 91, 72, 80, 77, 93, 69, 75, 79, 70, 83, 94, 75, 88

What is sample mean?  
### **ANSWER: 79.73**

In [3]:
data = [87, 63, 91, 72, 80, 77, 93, 69, 75, 79, 70, 83, 94, 75, 88]
n = len(data) # количество значений
summary_scores = sum(data)
sample_mean = summary_scores/n
sample_mean = np.round(sample_mean, 2)
print(f'sample mean = {sample_mean}')

sample mean = 79.73


3. The following data give the annual average number of inches of precipitation and the average number of days of precipitation in a sample of cities

(a) Find the sample mean of the average number of inches of precipitation.  
### **ANSWER: 33.0 inches**  
(b) Find the sample mean of the average number of days of precipitation.  
### **ANSWER: 102.38 days**  

In [4]:
precipitation_data = [35.74, 31.50, 11.43, 15.31, 19.59, 44.76, 47.29, 12.08, 57.55, 59.74, 36.30, 29.13, 28.61]
days_data = [134, 83, 95, 88, 100, 105, 127, 36, 129, 114, 154, 81, 85]

n_precipitation = len(precipitation_data)
n_days = len(days_data)

sum_precipitation = sum(precipitation_data)
sum_days = sum(days_data)

precipitation_sample_mean = np.round(sum_precipitation/n_precipitation, 2) # округление до сотых
days_sample_mean = np.round(sum_days/n_days, 2)

print(f'precipitation sample mean = {precipitation_sample_mean} inches')
print(f'days sample mean = {days_sample_mean} days')

precipitation sample mean = 33.0 inches
days sample mean = 102.38 days


5. The sample mean of the weights of the adult women of town A is larger than the sample mean of the weights of the adult women of town B. Moreover, the sample mean of the weights of the adult men of town A is larger than the sample mean of the weights of the adult men of town B. Can we conclude that the sample mean of the weights of the adults of town A is larger than the sample mean of the weights of the adults of town B? Explain your answer

### **ANSWER: No, because it depends on the number of people in the samples**

7. The following table gives the number of cases of tetanus in the
27-country European community in the years from 1996 to 2004. 

Find the sample mean of these 27 × 11 = 297 data values.  
### **ANSWER: 8.09**

In [5]:
data = [352, 309, 290, 288, 263, 220, 188, 219, 177, 98]
sum_data = sum(data)
n = 27*11
sample_mean = np.round(sum_data/n, 2)
print(f'sample mean = {sample_mean}')

sample mean = 8.09


9. Find the sample mean for this data set:
1, 2, 4, 7, 10, 12 **ANSWER: 6**

Now find the sample means for the data sets
3, 6, 12, 21, 30, 36 and 6, 7, 9, 12, 15, 17  
### **ANSWER: 18, 11**

In [6]:
first_data = [1, 2, 4, 7, 10, 12]
second_data = [3, 6, 12, 21, 30, 36]
third_data = [6, 7, 9, 12, 15, 17]

first_sm = np.round(sum(first_data)/len(first_data), 2)
second_sm = np.round(sum(second_data)/len(second_data), 2)
third_sm = np.round(sum(third_data)/len(third_data), 2)

print(f'first sample mean = {first_sm}')
print(f'second sample mean = {second_sm}')
print(f'third sample mean = {third_sm}')

first sample mean = 6.0
second sample mean = 18.0
third sample mean = 11.0


11. The following data give the total number of fires in Ontario, Canada,
in the months of 2002:

6, 13, 5, 7, 7, 3, 7, 2, 5, 6, 9, 8


Find the sample mean of this data set  
### **ANSWER: 6.5**

In [7]:
data = [6, 13, 5, 7, 7, 3, 7, 2, 5, 6, 9, 8]
sum_data = sum(data)
n = len(data)

sm = sum_data/n
print(f'sample mean = {sm}')

sample mean = 6.5


13. One-half the values of a sample are equal to 10, and the other half are equal to 20. What is the sample mean?

sum of data set equal to ((n/2)*(10+20))/n = n(10+20)/2n = (10+20)/2 = 15

### **ANSWER: sample mean = 15**

15. Half the values of a sample are equal to 10, one-sixth are equal to 20, and one-third are equal to 30. What is the sample mean?  
### **ANSWER: 18.(3)**

In [8]:
first_sm = 10
second_sm = 20
third_sm = 30

part1 = 1/2
part2 = 1/6
part3 = 1/3

total_sm = part1*first_sm + part2*second_sm + part3*third_sm

print(f'total sample mean = {total_sm}')

total sample mean = 18.333333333333332


17. A company runs two manufacturing plants. A sample of 30 engineers at plant 1 yielded a sample mean salary of $33,600. A sample of 20 engineers at plant 2 yielded a sample mean salary of $42,400. What is the sample mean salary for all 50 engineers?  
### **ANSWER: 37.120$**

In [9]:
n_sample1 = 30
n_sample2 = 20
n = n_sample1 + n_sample2
sm1 = 33600
sm2 = 42400
part1 = n_sample1/n
part2 = n_sample2/n

total_sm = part1*sm1 + part2*sm2
print(f'sample mean for 50 engineers = {int(total_sm)}$')

sample mean for 50 engineers = 37120$


19. Find the deviations for each of the three data sets of Prob. 9, and verify your answers by showing that in each case the sum of the deviations is 0  
### **ANSWER: all is ZERO**

In [10]:
first_data = [1, 2, 4, 7, 10, 12]
second_data = [3, 6, 12, 21, 30, 36]
third_data = [6, 7, 9, 12, 15, 17]

# sample mean = sum / n
first_sm = np.round(sum(first_data)/len(first_data), 2) 
second_sm = np.round(sum(second_data)/len(second_data), 2)
third_sm = np.round(sum(third_data)/len(third_data), 2)

# инициализация отклонений до расчетов
first_deviation = 0
second_deviation = 0
third_deviation = 0

# суммируем разность среднего и каждого значения в выборке
for i in first_data:
    first_deviation += first_sm - i

for i in second_data:
    second_deviation += second_sm - i

for i in third_data:
    third_deviation += third_sm - i

print(f'first data devation = {first_deviation}')
print(f'second data devation = {second_deviation}')
print(f'third data devation = {third_deviation}')

first data devation = 0.0
second data devation = 0.0
third data devation = 0.0


# 3.3 Sample Median

1. The following are the total yardages of a sample of 12 municipal golf
courses:  
7040, 6620, 6050, 6300, 7170, 5990, 6330, 6780, 6540, 6690, 6200, 6830

(a) Find the sample median.  
**ANSWER: 6580**  
(b) Find the sample mean  
**ANSWER: 6545**

In [11]:
data = [7040, 6620, 6050, 6300, 7170, 5990, 6330, 6780, 6540, 6690, 6200, 6830]
median = np.median(data) # (a) сортируется и выбирается значение посередине при нечетн. кол-ве сумма двух серединных при четн. кол-ве
sm = np.mean(data) # (b) сумма, деленная на кол-во элементов
print(f'median = {median}')
print(f'sample mean = {sm}')

median = 6580.0
sample mean = 6545.0


3. If the median of the data set x(i), i = 1, ... , n, is 10, what is the median of
the data set 2x(i) + 3, i = 1, ... , n?  

### **The median will be equal to 2*10+3 = 23, since our previous element haven't changed his position and all elements have been changed to 2x(i)+3**

5. The following presents the male and female suicide rates per 100,000 population for a variety of countries.  

(a) Find the sample median of the male suicide rates.  
### **ANSWER: 22.0**
(b) Find the sample median of the female suicide rates.  
### **ANSWER: 8.1**
(c) Find the sample mean of the male suicide rates.  
### **ANSWER: 23.68**
(d) Find the sample mean of the female suicide rates.  
### **ANSWER: 9.68**

In [16]:
males = [19.7, 18.2, 42.1, 20.5, 35.1, 33.1, 11.0, 27.8, 14.6, 22.0, 25.0, 12.1, 26.6]
females = [5.4, 5.1, 15.8, 5.4, 20.6, 12.7, 4.3, 14.9, 8.1, 4.4, 11.5, 5.7, 12.0]

m_median = np.round(np.median(males), 2)
f_median = np.round(np.median(females), 2)
m_sm = np.round(np.mean(males), 2)
f_sm = np.round(np.mean(females), 2)

print(f'males median = {m_median}')
print(f'females median = {f_median}')
print(f'males sample mean = {m_sm}')
print(f'females sample mean = {f_sm}')

males median = 22.0
females median = 8.1
males sample mean = 23.68
females sample mean = 9.68


7. Find the sample median of the average annual number of inches of precipitation in the cities noted in Prob. 3 of Sec. 3.2.  
### **ANSWER: 31.5**

In [109]:
precipitation_data = [35.74, 31.50, 11.43, 15.31, 19.59, 44.76, 47.29, 12.08, 57.55, 59.74, 36.30, 29.13, 28.61]
median = np.median(precipitation_data)
print(f'median = {median}')

median = 31.5


9. Use the table on death rates preceding Prob. 9 of Sec. 2.3 to find the sample median of the death rates due to  

(a) Falls  
### **ANSWER: 5.15**
(b) Poisoning  
### **ANSWER: 3.45**
(c) Drowning  
### **ANSWER: 1.5**

In [20]:
falls_data = [8.3, 5.9, 5.0, 4.9, 5.0, 5.0, 5., 5.2, 5.3, 5.6, 5.8, 6.0, 4.8, 4.8, 5.1, 5.2]
poisoning_data = [2.6, 1.9, 2.2, 2.3, 2.6, 2.7, 3.4, 3.5, 3.4, 3.5, 3.8, 4.0, 4.5, 4.6, 5.0, 5.6]
drowning_data = [3.9, 3.2, 2.2, 1.9, 1.8, 1.4, 1.5, 1.5, 1.7, 1.5, 1.5, 1.6, 1.3, 1.3, 1.2, 1.1]

# я мог отсортировать и выбрать медиану по середине, но мне оч лень, Гаухар Сериковна(
falls_median = np.round(np.median(falls_data), 2)
poisoning_median = np.round(np.median(poisoning_data), 2)
drowning_median = np.round(np.median(drowning_data), 2)

print(f'falls median = {falls_median}')
print(f'poisoning median = {poisoning_median}')
print(f'drowning median = {drowning_median}')

falls median = 5.15
poisoning median = 3.45
drowning median = 1.5


11. The histogram in the figure on the following page describes the annual rainfall, in inches, over the last 34 years in a certain western city. Since the raw data are not recoverable from a histogram, we cannot use them to exactly compute the value of the sample mean and sample median. Still, based on this histogram,  
what is the largest possible value of   

(a) The sample mean?  
### **ANSWER: 20.74**    
(b) The sample median?  
### **ANSWER: 20.5**  
What is the smallest possible value of  
(c) The sample mean?  
### **ANSWER: 19.74**  
(d) The sample median?  
### **ANSWER: 19.5**  
(e) The actual data follow:  
### **ANSWER: sample mean = 20.21, median = 20.05**  
[15.2, 16.1, 16.5, 16.7, 17.2, 17.5, 17.7, 18.3, 18.6, 18.8, 18.9, 19.1,  
19.2, 19.2, 19.6, 19.8, 19.9, 20.2, 20.3, 20.3, 20.8, 21.1, 21.4, 21.7,  
22.2, 22.5, 22.5, 22.7, 22.9, 23.3, 23.6, 24.1, 24.5, 24.9]  
Determine the sample mean and sample median and see that they are
consistent with your previous answers

In [46]:
hist_data = []
hist_values = list(range(15, 25)) # вернет список [15, 16, ..., 23, 24]
hist_counts = [1, 3, 3, 4, 6, 4, 3, 5, 2, 3] # количество вхождений (индексы соответствуют индексам значений)

# сформируем предположительную исходную выборку в целых числах
for i in range(len(hist_values)):
    for j in range(hist_counts[i]):
        hist_data.append(hist_values[i])

hist_sm = np.round(sum(hist_data)/len(hist_data), 2)
hist_median = np.round(np.median(hist_data), 2) 

print(f'smallest possible hist sample mean = {hist_sm}') # предполагаем, что все значения целочисленные — X.0
print(f'largest possible hist sample mean = {hist_sm + 1}') # предполагаем, что числа после запятой максимально велики — X.(9)

print(f'smallest possible hist median = {hist_median}')
print(f'largest possible hist median = {hist_median+1}')

data = [15.2, 16.1, 16.5, 16.7, 17.2, 17.5, 17.7, 18.3, 
    18.6, 18.8, 18.9, 19.1, 19.2, 19.2, 19.6, 19.8, 19.9, 
    20.2, 20.3, 20.3, 20.8, 21.1, 21.4, 21.7, 22.2, 22.5, 
    22.5, 22.7, 22.9, 23.3, 23.6, 24.1, 24.5, 24.9]

# основываясь на фактических данных
data_sm = np.round(np.mean(data), 2)
data_median = np.round(np.median(data), 2)

print(f'sample mean = {data_sm}')
print(f'median = {data_median}')


smallest possible hist sample mean = 19.74
largest possible hist sample mean = 20.74
smallest possible hist median = 19.5
largest possible hist median = 20.5
sample mean = 20.21
median = 20.05


13. Using the data from Example 3.4, compute the sample medians of the severity of head injuries suffered by motorcycle operators who were wearing and who were not wearing helmets.  
### **ANSWER: The value of both medians is equal 0, because >= 50% of observations is 0**

15. Women make up the following percentages of the workforce in the 14 occupations listed.
For these percentages find  
(a) The sample mean **ANSWER: 32.52**  
(b) The sample median **ANSWER: 24.25**  
It also turns out that women make up 44.4 percent of the total workforce for these occupations. Is this consistent with your answers in (a) and (b)? Explain!  
### **ANSWER: As in one of the previous problems with weights, it also depends on number of values in sample**

In [54]:
data = [36.8, 94.3, 30.5, 68.6, 1.9, 41.5, 2.8, 17.6, 18.0, 85.2, 43.5, 10.9, 1.6, 2.1]
data_sm = np.round(np.mean(data_sm), 2)
data_median = np.round(np.median(data), 2)
print(f'sample mean = {data_sm}')
print(f'median = {data_median}')

sample mean = 32.52
median = 24.25


17. The following table gives the median age at first marriage in the years 1992 to 2002.  
(a) Find the sample median of the men’s median age.  
### **ANSWER: 26.8**  
(b) Find the sample median of the women’s median age.  
### **ANSWER: 25.0**  

In [56]:
women = [25.3, 25.1, 25.1, 25.1, 25.0, 25.0, 24.8, 24.5, 24.5, 24.5, 24.4]
men = [26.9, 26.9, 26.8, 26.9, 26.7, 26.8, 27.1, 26.9, 26.7, 26.5, 26.5]

w_median = np.round(np.median(women), 2)
m_median = np.round(np.median(men), 2)

print(f'women median = {w_median}')
print(f'men median = {m_median}')

women median = 25.0
men median = 26.8


1. Match each statement in the left-hand column with the correct data set
from the right-hand column.  

1) Sample mode is 9  
2) Sample mean is 9  
3) Sample median is 9   
A) 5, 7, 8, 10, 13, 14   
B) 1, 2, 5, 9, 9, 15  
C) 1, 2, 9, 12, 12, 18  

### **ANSWER: 1-B, 2-C, 3-A**

### пришлось парсить книжку и запихивать данные в эксель. могу за поинты сделать еще =)) во благо лени
3. Using data concerning the first 100 students in App. A, find the sample mode for  
(a) Weight  
### **ANSWER: [126]**  
(b) Blood pressure  
### **ANSWER: [110, 102, 114]**  
(c) Cholesterol  
### **ANSWER: [196]**  

In [95]:
data = pd.read_excel('./appendix_A.xlsx')
data.head(10)

Student  Weight  Cholesterol  Pressure Gender
0        1     147          213       127      F
1        2     156          174       116      M
2        3     112          193       110      F
3        4     127          196       110      F
4        5     144          220       130      F
5        6     140          183        99      M
6        7     119          194       112      F
7        8     139          200       102      F
8        9     161          192       121      M
9       10     146          200       125      F

In [107]:
weight_data = data['Weight'][0:100]
pressure_data = data['Pressure'][0:100]
cholesterol_data = data['Cholesterol'][0:100]


weight_mode = statistics.multimode(weight_data)
pressure_mode = statistics.multimode(pressure_data)
cholesterol_mode = statistics.multimode(cholesterol_data)

print(f'weight modes = {weight_mode}')
print(f'pressure modes = {pressure_mode}')
print(f'cholesterol modes = {cholesterol_mode}')

weight modes = [126]
pressure modes = [110, 102, 114]
cholesterol modes = [196]


5. Construct a data set for which the sample mean is 10, the sample median is 8, and the sample mode is 6.  
### **ANSWER: [6, 6, 6, 7, 9, 10, 12, 24] (below is a check)**

In [108]:
data = [6, 6, 6, 7, 9, 10, 12, 24]
sm = np.mean(data)
median = np.median(data)
mode = statistics.mode(data)

print(f'sample mean = {sm}')
print(f'sample median = {median}')
print(f'sample mode = {mode}')

sample mean = 10.0
sample median = 8.0
sample mode = 6


7. Joggers use a quarter-mile track around an athletic field. In a sample of 17 joggers, 1 did 2 loops, 4 did 4 loops, 5 did 6 loops, 6 did 8 loops, and 1 did 12 loops.  
(a) What is the sample mode of the number of loops run by these joggers?  
### **ANSWER: mode is <u>8 loops</u>**  
(b) What is the sample mode of the distances run by these joggers?  
### **ANSWER: mode is 8 * 0.25 = <u>2 miles</u>**